src/controller/java/AndroidConnectionFailureExceptions.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "AndroidConnectionFailureExceptions.h"
+
+#include <lib/core/CHIPError.h>
+#include <lib/support/CHIPJNIError.h>
+#include <lib/support/JniReferences.h>
+#include <lib/support/JniTypeWrappers.h>
+
+namespace chip {
+
+CHIP_ERROR AndroidConnectionFailureExceptions::CreateAndroidConnectionFailureException(JNIEnv * env, const char * message,
+                                                                                       uint32_t errorCode,
+                                                                                       SessionEstablishmentStage state,
+                                                                                       jthrowable & outEx)
+{
+    jclass controllerExceptionCls;
+    CHIP_ERROR err =
+        JniReferences::GetInstance().GetClassRef(env, "chip/devicecontroller/ConnectionFailureException", controllerExceptionCls);
```

Comment:
Please use GetLocalClassRef here, this would be callback, and is called from c++, we have used GetLocalClassRef pattern in the callback flow and avoid the leak
https://github.com/project-chip/connectedhomeip/blob/d0d95d319016b07b6b0cd80742ac28b7593fc51a/src/lib/support/JniReferences.h#L81

Reply:
you may reference https://github.com/project-chip/connectedhomeip/blob/8f29cacec025dccb19fe65fe596691f26c4b1647/src/controller/java/AndroidControllerExceptions.cpp#L27

And https://github.com/project-chip/connectedhomeip/blob/8f29cacec025dccb19fe65fe596691f26c4b1647/src/controller/java/AndroidClusterExceptions.cpp is problematic, no one use it, so I have not yet updated the pattern there, we would delete this deadcode  soon.